homeassistant/components/climate/opentherm_gw.py

```diff
@@ -0,0 +1,201 @@
+            self._away_mode_a = 1
+        else:
+            self._away_mode_a = None
+        if status.get(self.pyotgw.OTGW_GPIO_B) == 5:
+            self._away_mode_b = 0
+        elif status.get(self.pyotgw.OTGW_GPIO_B) == 6:
+            self._away_mode_b = 1
+        else:
+            self._away_mode_b = None
+        if self._away_mode_a is not None:
+            self._away_state_a = (status.get(self.pyotgw.OTGW_GPIO_A_STATE) ==
+                                  self._away_mode_a)
+        if self._away_mode_b is not None:
+            self._away_state_b = (status.get(self.pyotgw.OTGW_GPIO_B_STATE) ==
+                                  self._away_mode_b)
+        self.hass.async_add_job(self.async_update_ha_state(True))
+
+    @property
+    def name(self):
+        """Return the friendly name."""
+        return self.friendly_name
+
+    @property
+    def state(self):
+        """Return the current state."""
+        if self.is_on is False:
+            return STATE_OFF
+        if self._current_operation:
+            return self._current_operation
+        if self.is_on:
+            return STATE_ON
+        return STATE_UNKNOWN
+
+    @property
+    def precision(self):
+        """Return the precision of the system."""
+        if self.temp_precision is not None:
+            return self.temp_precision
+        if self.unit_of_measurement == TEMP_CELSIUS:
+            return PRECISION_HALVES
+        return PRECISION_WHOLE
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement to display."""
+        return self.hass.config.units.temperature_unit
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement used by the platform."""
+        return TEMP_CELSIUS
+
+    @property
+    def current_operation(self):
+        """Return current operation ie. heat, cool, idle."""
+        return self._current_operation
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        if self.floor_temp is True:
+            if self.temp_precision == PRECISION_HALVES:
+                return int(2 * self._current_temperature) / 2
+            if self.temp_precision == PRECISION_TENTHS:
+                return int(10 * self._current_temperature) / 10
+            return int(self._current_temperature)
+        return self._current_temperature
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return self._target_temperature
+
+    @property
+    def target_temperature_step(self):
+        """Return the supported step of target temperature."""
+        return self.temp_precision
+
+    @property
+    def is_away_mode_on(self):
+        """Return true if away mode is on."""
+        return (getattr(self, '_away_state_a', False) or
+                getattr(self, '_away_state_b', False))
+
+    def async_set_temperature(self, **kwargs):
```

Comment:
This should be a coroutine.

Reply:
addressed in c5a9266871b32a981ad19ab38a0bad650ab644e4